packages/k8s/src/k8s/index.ts

```diff
@@ -326,7 +323,14 @@ export async function waitForPodPhases(
   let phase: PodPhase = PodPhase.UNKNOWN
   try {
     while (true) {
-      phase = await getPodPhase(podName)
+      try {
+        phase = await getPodPhase(podName)
+      } catch (err) {
+        const e = err as k8s.HttpError
```

Comment:
This is the case when we are creating for example a `StatefulSet` for `docker-registry`. The controller is about to create a pod, but we can hit this function potentially before the reconciliation happens. I don't think that we should handle it this way, I just made it like this until we figure out how to proceed.

One way is to add retries in this case. The other solution is to have pod listing of the outer resource, so we wait there before we actually await for the pod phase. Not sure which one should be more appropriate. Maybe there is a better way that I am not seeing right now :relaxed: